### PR TITLE
Wasm32 support

### PR DESCRIPTION
**Pull request by yzsolt**
Adds support for WebAssembly SIMD, using the currently available stable intrinsics in Rust.

I also added a GitHub action for automated testing. Benchmarking also works, and shows a ~3x improvement of SIMD vs. scalar when running in `wasmtime`.

Note that I'm not a SIMD expert, so I may have missed some optimization opportunities when implementing operations which don't map 1:1 to WASM SIMD intrinsics. I've tried to follow already existing SSE and NEON implementations though.

Closes https://github.com/arduano/simdeez/issues/20.

### CODE REVIEW

File: src/tests/lib/tester.rs [hunk 1]
```diff
@@ -229,11 +233,12 @@ macro_rules! elementwise_eq_tester_impl {
     };
 
     (@simdkind $simd_ty:ident, $simd_base:ident, $simd_fn:ident, $arg_cnt:ident, $precision:expr) => {
-        elementwise_eq_tester_impl!(@full Scalar, $simd_ty, $simd_base, $simd_fn, $arg_cnt, $precision);
+        //elementwise_eq_tester_impl!(@full Scalar, $simd_ty, $simd_base, $simd_fn, $arg_cnt, $precision);
```
Comment (arduano) [inline src/tests/lib/tester.rs:236]: Please keep scalar uncommented

**Review by arduano — CHANGES_REQUESTED** [review]

**Comment by arduano** [comment]
Request my review when you think it's ready, also please mention how to run/test it

**Comment by yzsolt** [comment]
Hey, I think it's ready! Rebased onto master and reenabled the commented out scalar tests.

About testing: I've set up things in a way that if you specify `wasm32-wasi` as target, `wasmtime` (de-facto standard Rust WebAssembly runtime) will be used for running the produced .wasm modules. It's installed on CI automatically, locally you have [to do it yourself](https://docs.wasmtime.dev/cli-install.html) beforehand. Both `cargo test` and `cargo bench` work for `wasm32-wasi`. `cargo bench` spits out some error logs (I think it's related to limited file access inside WebAssembly), but otherwise works.

For more complex testing you can build `simdnoise` for WebAssembly, which I did back when I initially implemented this. It worked nicely, got the same 2-3x scalar -> SIMD speedups in WASM as in native code.

**Review by arduano — APPROVED** [review]
Looks about right, I can't find any issues. Is it good to merge?

**Comment by yzsolt** [comment]
> Looks about right, I can't find any issues. Is it good to merge?

Yes, ship it!

**Pull request merged**